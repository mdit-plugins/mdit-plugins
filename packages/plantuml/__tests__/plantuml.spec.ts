import MarkdownIt from "markdown-it";
import { expect, it } from "vitest";

import { plantuml } from "../src/index.js";

it("should render without options", () => {
  const markdownIt = MarkdownIt({ linkify: true }).use(plantuml);

  const result = markdownIt.render(`
@startuml
Bob -> Alice : hello
@enduml
    `);

  expect(result).toMatchSnapshot();
  expect(result).toMatch(/<img src=".*" alt=".*">/);
});

it("should render now-ascii characters", () => {
  const markdownIt = MarkdownIt({ linkify: true }).use(plantuml);

  const result = markdownIt.render(`
@startuml
Bob -> Alice : 你好
@enduml
    `);

  expect(result).toMatchSnapshot();
  expect(result).toMatch(/<img src=".*" alt=".*">/);
});

it("should parse custom uml", () => {
  const markdownIt = MarkdownIt({ linkify: true }).use(plantuml, {
    name: "json",
  });

  const result = markdownIt.render(`
@startjson
{
  "firstName": "John",
  "lastName": "Smith",
  "isAlive": true,
  "age": 27,
  "address": {
    "streetAddress": "21 2nd Street",
    "city": "New York",
    "state": "NY",
    "postalCode": "10021-3100"
  },
  "phoneNumbers": [
    {
      "type": "home",
      "number": "212 555-1234"
    },
    {
      "type": "office",
      "number": "646 555-4567"
    }
  ],
  "children": [],
  "spouse": null
}
@endjson
    `);

  expect(result).toMatchSnapshot();
  expect(result).toMatch(/<img src=".*" alt=".*">/);
});

it("should parse code block", () => {
  const markdownIt = MarkdownIt({ linkify: true }).use(plantuml, {
    type: "fence",
  });

  const result = markdownIt.render(`
\`\`\`uml
Bob -> Alice : hello
\`\`\`
    `);

  expect(result).toMatchSnapshot();
  expect(result).toMatch(/<img src=".*" alt=".*">/);
});

it("should parse custom code block", () => {
  const markdownIt = MarkdownIt({ linkify: true }).use(plantuml, {
    type: "fence",
    name: "json",
    fence: "jsonuml",
  });

  const result = markdownIt.render(`
\`\`\`jsonuml
{
  "firstName": "John",
  "lastName": "Smith",
  "isAlive": true,
  "age": 27,
  "address": {
    "streetAddress": "21 2nd Street",
    "city": "New York",
    "state": "NY",
    "postalCode": "10021-3100"
  },
  "phoneNumbers": [
    {
      "type": "home",
      "number": "212 555-1234"
    },
    {
      "type": "office",
      "number": "646 555-4567"
    }
  ],
  "children": [],
  "spouse": null
}
\`\`\`
    `);

  expect(result).toMatchSnapshot();
  expect(result).toMatch(/<img src=".*" alt=".*">/);
});

it("should not break normal code block", () => {
  const markdownIt = MarkdownIt({ linkify: true }).use(plantuml, {
    type: "fence",
  });

  const result = markdownIt.render(`
\`\`\`js
const a = 1
\`\`\`
    `);

  expect(result).toMatchSnapshot();
  expect(result).toMatch(/<pre><code[\s\S]*<\/code><\/pre>/);
});
