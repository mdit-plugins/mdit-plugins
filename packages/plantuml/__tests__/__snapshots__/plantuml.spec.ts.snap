// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should not break normal code block 1`] = `
"<pre><code class="language-js">const a = 1
</code></pre>
"
`;

exports[`should parse code block 1`] = `
"<img src="https://www.plantuml.com/plantuml/svg/SoWkIImgAStDuNBAJrBGjLDmpCbCJbMmKiX8pSd9vt98pKifpSq10000" alt="plantuml diagram">
"
`;

exports[`should parse custom code block 1`] = `
"<img src="https://www.plantuml.com/plantuml/svg/RP6nRiCW441tlk9qSoBPn7QaJ8cwTV2IAQewK8DhMWmMd5j5bV-zufoqIqRUSOydSKoa8dsauCMF0C3UnaIj6WqU09_3u76JiLF_z3nQ6bPiqydPhqmfpeQHUiz7kLyFMaUJ4XuWom4mKJI6JdyOPGdIQpWpPoa0TfQkUTYQRxY4-7ddYHHnGdkveoaaKkufQEPbKSXokokB0WN0mWtJ4Bnfv_7DnFpY2--jEG18ruatXp2QcnC0FLzV-oJKTRqjvQv27Y-RnuRGzxPxw6YgXXrLtUnl3W7modNTO9sEnkUobQGfp2aR_UoSMCJHU9t_vnS0" alt="plantuml diagram">
"
`;

exports[`should parse custom uml 1`] = `
"<img src="https://www.plantuml.com/plantuml/svg/RP6nRiCW441tlk9qSoBPn7QaJ8cwTV2IAQewK8DhMWmMd5j5bV-zufoqIqRUSOydSKoa8dsauCMF0C3UnaIj6WqU09_3u76JiLF_z3nQ6bPiqydPhqmfpeQHUiz7kLyFMaUJ4XuWom4mKJI6JdyOPGdIQpWpPoa0TfQkUTYQRxY4-7ddYHHnGdkveoaaKkufQEPbKSXokokB0WN0mWtJ4Bnfv_7DnFpY2--jEG18ruatXp2QcnC0FLzV-oJKTRqjvQv27Y-RnuRGzxPxw6YgXXrLtUnl3W7modNTO9sEnkUobQGfp2aR_UoSMCJHU9t_vnS0" alt="">
"
`;

exports[`should render now-ascii characters 1`] = `
"<img src="https://www.plantuml.com/plantuml/svg/SoWkIImgAStDuNBAJrBGjLDmpCbCJbMmKdYoTy7JfNkv75BpKafpSm00" alt="">
"
`;

exports[`should render without options 1`] = `
"<img src="https://www.plantuml.com/plantuml/svg/SoWkIImgAStDuNBAJrBGjLDmpCbCJbMmKiX8pSd9vt98pKifpSq10000" alt="">
"
`;
